test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/getThreadListStackTracesExtended/gtlste002.java

```diff
@@ -0,0 +1,119 @@
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+package com.ibm.jvmti.tests.getThreadListStackTracesExtended;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.locks.LockSupport;
+
+public class gtlste002 {
+	static long ii = 0;
+	static final int iter = 1000;
+	static int reachedIter = 0;
+
+	/**
+	 * Check if there is at least one JITted frame in the stack trace of the given thread.
+	 * @param t the thread to be checked, if t is null, the current thread will be used
+	 * @return if there is at least one JITted frames in the stack trace of the given thread
+	 */
+	public static native boolean anyJittedFrame(Thread t);
+
+	/**
+	 * Method to be compiled in the tests. If the virtual thread will be unmounted, the
+	 * return value should be ignored as it will be checked separately.
+	 * @param mounted if the tested virtual thread will be mounted or not
+	 * @return if there is at least one JITted frames in the stack trace of the virtual thread.
+	 */
+	boolean methodToCompile(boolean mounted) {
+		for (int j = 0; j < 1000; j++) {
+			ii++;
+		}
+
+		if (reachedIter == iter) {
+			if (mounted) {
+				return anyJittedFrame(null);
+			} else {
+				LockSupport.park();
+			}
+		}
+		return true;
+	}
+
+	public String helpUnmountedVthread() {
+		return "Test jvmtiGetThreadListStackTracesExtended for at least 1 jitted frame after a relatively large number of iterations on unmounted virtual threads.";
+	}
+
+	public boolean testUnmountedVthread() {
+		boolean rc = false;
+		reachedIter = 0;
+
+		try {
+			Thread t1 = Thread.ofVirtual().name("vthread 1").start(() -> {
+				while (reachedIter < iter) {
+					reachedIter++;
+					methodToCompile(false);
+				}
+			});
+
+			while (t1.getState() != Thread.State.WAITING) {
+				Thread.sleep(500);
+			}
+			rc = anyJittedFrame(t1);
+
+			LockSupport.unpark(t1);
+
+			t1.join();
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+
+		return rc;
+	}
+
+	public String helpMountedVthread() {
+		return "Test jvmtiGetThreadListStackTraceExtended for at least 1 jitted frame after a relatively large number of iterations on mounted virtual threads.";
+	}
+
+	public boolean testMountedVthread() {
```

Comment:
Can you also describe how the test suite invokes variants of `helpMountedVthread` and `helpMountedVthread`?

Reply:
Any function whose name starts with "test" will be called by the test suite.